src/tools/rbd_mirror/image_map/Policy.h

```diff
@@ -0,0 +1,73 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_POLICY_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_POLICY_H
+
+#include <tuple>
+#include <vector>
+
+#include "types.h"
+#include "common/RWLock.h"
+#include "tools/rbd_mirror/types.h"
+#include "include/rados/librados.hpp"
+
+using librados::IoCtx;
+
+class Context;
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+class Policy {
+public:
+  Policy(IoCtx &ioctx);
+  virtual ~Policy() {
+  }
+
+  // interfaces for anyone using the map
+
+  // init (load) the instance map from disk
+  void init(Context *on_finish);
+
+  // handle additional/removal of images
+  MapSpecRef handle_images(
```

Comment:
This really should be returning ```void``` since you will need an async callback to a listener.